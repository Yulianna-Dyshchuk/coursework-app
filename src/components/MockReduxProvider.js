import React from "react";
import { combineReducers, createStore } from "redux";
import { Provider } from "react-redux";

const store = createStore(combineReducers({ categories: () => ([]) }));

export const MockReduxProvider = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
