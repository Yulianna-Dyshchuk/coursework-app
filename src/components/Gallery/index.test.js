import React from "react";
import { shallow } from "enzyme";
import { Gallery } from "./index";
import { MockReduxProvider } from "../MockReduxProvider";

describe("Gallery component", () => {
  it("should render correctly without props", () => {
    const component = shallow(
      <MockReduxProvider>
        <Gallery />
      </MockReduxProvider>
    );

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with props", () => {
    const component = shallow(
      <MockReduxProvider>
        <Gallery gallery={[]} category="all" />
      </MockReduxProvider>
    );

    expect(component).toMatchSnapshot();
  });
});
