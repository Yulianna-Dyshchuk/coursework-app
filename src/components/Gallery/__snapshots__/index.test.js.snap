// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component should render correctly with props 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Gallery
    category="all"
    gallery={Array []}
  />
</Provider>
`;

exports[`Gallery component should render correctly without props 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Gallery />
</Provider>
`;
